### PR TITLE
feat: make the demo_client work without Docker

### PR DESCRIPTION
**Pull request by huitseeker**
This create a local scenario that runs the testbed and launches the demo against it. 
It succeeds in retrieving collections as intended.

### CODE REVIEW

File: benchmark/benchmark/full_demo.py [hunk 1]
```diff
@@ -0,0 +1,154 @@
+# Copyright(C) Facebook, Inc. and its affiliates.
```
Comment (arun-koshy) [inline benchmark/benchmark/full_demo.py:1]: nit: update license to include mysten

File: examples/src/demo_client.rs [hunk 2]
```diff
@@ -1,14 +1,13 @@
 // Copyright (c) 2022, Mysten Labs, Inc.
 // SPDX-License-Identifier: Apache-2.0
-#[allow(unused_imports)]
-use narwhal::configuration_client::ConfigurationClient;
+
+use clap::{crate_name, crate_version, App, AppSettings, SubCommand};
+// use narwhal::configuration_client::ConfigurationClient;
```
Comment (arun-koshy) [inline examples/src/demo_client.rs:5]: nit: remove this comment?

File: benchmark/benchmark/full_demo.py [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+    def _kill_nodes(self):
+        try:
+            cmd = CommandMaker.kill().split()
+            subprocess.run(cmd, stderr=subprocess.DEVNULL)
+        except subprocess.SubprocessError as e:
+            raise BenchError('Failed to kill testbed', e)
+
+    def run(self, debug=False):
+        assert isinstance(debug, bool)
+        Print.heading('Starting local benchmark')
```
Comment (arun-koshy) [inline benchmark/benchmark/full_demo.py:45]: Update print to `Starting local demo`

**Review by arun-koshy — APPROVED** [review]
Thanks for this! Can you update the comment [here](https://github.com/MystenLabs/narwhal/blob/e7636633a68ebd4d0d92ba243f3de1a84ceea19e/examples/src/demo_client.rs#L156) to reflect the current status of retrieving collections only working in the non docker case

**Comment by arun-koshy** [comment]
Can you also update the [README](https://github.com/MystenLabs/narwhal/blob/main/examples/README.md) please 🙏🏽

**Review by akichidis — APPROVED** [review]
Awesome work @huitseeker !! 💯

**Pull request merged**